vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,477 @@
+  is_step_ahead = TRUE,
+  time_unit = "week"
+)
+
+# Combine the `target_metadata_item` objects to create a `target_metadata`
+# class object, the `target_metadata_example`
+target_metadata_example <- create_target_metadata(target_metadata_hosp, target_metadata_death)
+```
+
+If you inspect the `target_metadata_hosp` object, you will see that it appears
+as a class `target_metadata_item` with additional attributes about the schema
+that created it:
+
+```{r tmh}
+str(target_metadata_hosp)
+```
+
+
+Likewise, `target_metadata_example` is a combination of `target_metadata_hosp`,
+and `target_metadata_death`:
+
+
+```{r, tme}
+str(target_metadata_example)
+```
+
+
+If this creates lists with metadata, then why bother creating a function at all?
+Ah, but you forget that people are fallible and it's easy to make mistakes. For
+example, some of the values are interdependent and if you accidentally leave one
+out, the function will provide a helpful error:
+
+```{r tmh-example}
+#| error: true
+create_target_metadata_item(
+  target_id = "inc hosp",
+  target_name = "Weekly incident influenza hospitalizations",
+  target_units = "rate per 100,000 population",
+  target_keys = list(target = "inc hosp"),
+  target_type = "discrete",
+  is_step_ahead = TRUE
+)
+```
+
+Now that we have defined the targets in `target_metadat_example`, we can move on
+to defining the task ID objects, which will define what the modeling parameters
+will be.
+
+### Creating the `task_id` objects
+
+In this modeling effort, we want to **require** modelers to provide week-ahead
+predictions for the "weekly incident influenza hospitalizations" target
+(`inc hosp`). Model submissions can optionally include five states, provide up
+to 4-week-ahead predictions, and also provide predictions for the "weekly
+incident influenza deaths" target (`inc death`).
+
+```{r set-task-ids}
+
+# Create a representation of a task ID item as a list object of `class
+# task_id`, the `origin_date_id`
+origin_date_id <- create_task_id(
+  "origin_date",
+  required = NULL,
+  optional = c("2023-01-02", "2023-01-09", "2023-01-16")
+)
+
+# Create a representation of a task ID item as a list object of `class
+# task_id`, the `location_id`
+location_id <- create_task_id(
+  "location",
+  required = "US",
+  optional = c("01", "02", "04", "05", "06")
+)
+
+# Create a representation of a task ID item as a list object of `class
+# task_id`, the `horizon_id`
+horizon_id <- create_task_id("horizon", required = 1L, optional = 2:4)
+
+# Create a representation of a task ID item as a list object of `class
+# task_id`, the `target_id`
+target_id <- create_task_id("target", required = "inc hosp", optional = "inc death")
+
+# Combine the `task_id` objects to create a `task_ids` class object, the
+# `task_ids_example`
+task_ids_example <- create_task_ids(origin_date_id, location_id, horizon_id, target_id)
+
+```
+
+Now that we've created the modeling parameters, we can define the outputs we
```

Comment:
Agreed. Since there's a double "the" in your suggestion, I'm going to add it here and I'm also going to de-duplicate "define"

```suggestion
Now that we've created the task IDs that specify the modeling tasks, we can define the outputs we
```